cartography/intel/azure/sql.py

```diff
@@ -277,267 +291,237 @@ def load_server_details(neo4j_session, credentials, subscription_id, details, up
+) -> None:
     """
     Ingest the elastic pool details into neo4j.
     """
     ingest_elastic_pools = """
-    MERGE (e:AzureElasticPool{id: {PoolId}})
+    UNWIND {elastic_pools_list} as ep
+    MERGE (e:AzureElasticPool{id: ep.id})
     ON CREATE SET e.firstseen = timestamp(), e.lastupdated = {azure_update_tag}
-    SET e.name = {Name},
-    e.location = {Location},
-    e.kind = {Kind},
-    e.creationdate = {CreationDate},
-    e.state = {State},
-    e.maxsizebytes = {MaxSizeBytes},
-    e.licensetype = {LicenseType},
-    e.zoneredundant = {ZoneRedundant}
-    WITH e
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET e.name = ep.name,
+    e.location = ep.location,
+    e.kind = ep.kind,
+    e.creationdate = ep.creation_date,
+    e.state = ep.state,
+    e.maxsizebytes = ep.max_size_bytes,
+    e.licensetype = ep.license_type,
+    e.zoneredundant = ep.zone_redundant
+    WITH e, ep
+    MATCH (s:AzureSQLServer{id: ep.server_id})
     MERGE (s)-[r:RESOURCE]->(e)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {azure_update_tag}
     """
 
-    for elastic_pool in elastic_pools:
-        neo4j_session.run(
-            ingest_elastic_pools,
-            PoolId=elastic_pool['id'],
-            Name=elastic_pool['name'],
-            Location=elastic_pool['location'],
-            Kind=elastic_pool['kind'],
-            CreationDate=elastic_pool['creation_date'],
-            State=elastic_pool['state'],
-            MaxSizeBytes=elastic_pool['max_size_bytes'],
-            LicenseType=get_optional_value(elastic_pool, ['license_type']),
-            ZoneRedundant=elastic_pool['zone_redundant'],
-            ServerId=elastic_pool['server_id'],
-            azure_update_tag=update_tag,
-        )
+    neo4j_session.run(
+        ingest_elastic_pools,
+        elastic_pools_list=elastic_pools,
+        azure_update_tag=update_tag,
+    )
 
 
 @timeit
-def _load_databases(neo4j_session, databases, update_tag):
+def _load_databases(
+        neo4j_session: neo4j.Session, databases: List[Dict], update_tag: int
+) -> None:
     """
     Ingest the database details into neo4j.
     """
     ingest_databases = """
-    MERGE (d:AzureDatabase{id: {Id}})
+    UNWIND {databases_list} as az_database
+    MERGE (d:AzureSQLDatabase{id: az_database.id})
     ON CREATE SET d.firstseen = timestamp(), d.lastupdated = {azure_update_tag}
-    SET d.name = {Name},
-    d.location = {Location},
-    d.kind = {Kind},
-    d.creationdate = {CreationDate},
-    d.databaseid = {DatabaseId},
-    d.maxsizebytes = {MaxSizeBytes},
-    d.licensetype = {LicenseType},
-    d.secondarylocation = {SecondaryLocation},
-    d.elasticpoolid = {ElasticPoolId},
-    d.collation = {Collation},
-    d.failovergroupid = {FailoverGroupId},
-    d.restorabledroppeddbid = {RDDId},
-    d.recoverabledbid = {RecoverableDbId}
-    WITH d
-    MATCH (s:AzureServer{id: {ServerId}})
+    SET d.name = az_database.name,
+    d.location = az_database.location,
```

Comment:
location cannot be changed after a resource is created. so, let's move it to the create set for all resources in sql, storage, etc.

Reply:
Done